src/redux/reducers/workspaces-reducer.ts

```diff
@@ -27,7 +29,15 @@ const setWorkspaces = (state: WorkspacesStore, { payload }: { payload: { data: W
   isLoading: false,
 });
 
+const setWorkspace = (state: WorkspacesStore, { payload }: { payload: { data: Workspace } }) => ({
+  ...state,
+  selectedWorkspace: payload.data,
+  isLoading: false,
+});
+
 export default {
   [`${FETCH_WORKSPACES}_PENDING`]: setLoadingState,
   [`${FETCH_WORKSPACES}_FULFILLED`]: setWorkspaces,
+  [`${FETCH_WORKSPACE}_PENDING`]: setLoadingState,
```

Comment:
It might be better to have loading state for the detail as well
```suggestion
  [`${FETCH_WORKSPACE}_PENDING`]: setLoadingDetailState,
```